### PR TITLE
feat: add from_engine_sync

### PR DESCRIPTION
**Pull request by averikitsch**
* Add example of sync from engine method
* Clean up async/sync in tests

### CODE REVIEW

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ async def afrom_instance(
         )
 
     @classmethod
-    def from_engine(cls, engine: AsyncEngine) -> PostgresEngine:
+    async def afrom_engine(cls, engine: AsyncEngine) -> PostgresEngine:
```
Comment (kurtisvg) [inline src/langchain_google_cloud_sql_pg/engine.py:212]: This is a breaking change -- we need to make sure to update the commit message to something like: "fix!: rename from_engine to afrom_engine"

**Review by kurtisvg — APPROVED** [review]

File: src/langchain_google_cloud_sql_pg/engine.py (hunk 1 above)
Comment (jackwotherspoon) [reply src/langchain_google_cloud_sql_pg/engine.py:212]: Why do we need an `afrom_engine`, the initialization of a SQLAlchemy async engine is synchronous. Thus wouldn't we want to support `from_engine`?

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 2]
```diff
@@ -209,7 +209,11 @@ async def afrom_instance(
         )
 
     @classmethod
-    async def afrom_engine(cls, engine: AsyncEngine) -> PostgresEngine:
+    def from_engine_sync(cls, engine: AsyncEngine, loop) -> PostgresEngine:
+        return cls(engine, loop, None)
+
+    @classmethod
+    def from_engine(cls, engine: AsyncEngine) -> PostgresEngine:
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/engine.py:216]: is this supposed to be async?

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]
I still do not understand why we need two `from_engine` methods? There should be no need for an async `from_engine` since the init of an async engine is sync code...

File: src/langchain_google_cloud_sql_pg/engine.py (hunk 2 above)
Comment (averikitsch) [reply src/langchain_google_cloud_sql_pg/engine.py]: You are right, the one method can support both the sync and async environments. We need to save the loop in order for the engine to be able to use sync methods.

File: src/langchain_google_cloud_sql_pg/engine.py [hunk 3]
```diff
@@ -209,8 +209,12 @@ async def afrom_instance(
         )
 
     @classmethod
-    def from_engine(cls, engine: AsyncEngine) -> PostgresEngine:
-        return cls(engine, None, None)
+    def from_engine(
+        cls,
+        engine: AsyncEngine,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
```
Comment (jackwotherspoon) [inline src/langchain_google_cloud_sql_pg/engine.py:215]: I need to take a deep look at this and think it through fully. Nowadays in Python we are really trying to now have any users deal with `loop` params. It leads to all sorts of issues if the user has the ability to configure the event loop.

I wonder if we can maybe lazy init somehow. Like try `asyncio.get_running_loop()` and if it errors set `loop=None` and then later on when we can guarantee we are in async land we set loop.

I have to think about this design as its tricky.

**Pull request closed without merging**